Source/scrollrt.cpp

```diff
@@ -677,8 +715,12 @@ static void DrawPlayerHelper(const CelOutputBuffer &out, int x, int y, int sx, i
 	}
 
 	PlayerStruct *pPlayer = &plr[p];
-	int px = sx + pPlayer->position.offset.x - CalculateWidth2(pPlayer->_pAnimWidth);
-	int py = sy + pPlayer->position.offset.y;
+	Point offset = ScrollInfo.offset;
+	if (pPlayer->IsWalking()) {
+		offset = GetOffsetForWalking(pPlayer->AnimInfo, pPlayer->_pdir);

```

Comment:
this also needs to be fed into `CheckCursMove()` for mouse clicks to happen correctly.

Source/cursor.cpp::278

Reply:
Done. 🙂 
I checked that monsters are highlighted while walking.